### PR TITLE
Use addEventProcessor for compatibility with sentry 4 and 5

### PR DESCRIPTION
**Pull request by jardakotesovec**
Based on [example](https://github.com/getsentry/sentry-javascript/issues/1781#issuecomment-444510327) from sentry developer I used `addEventProcessor` to enrich the event.

If I don't miss something, this could be easy update which has compatibility with both sentry 4 and 5.

### CODE REVIEW

**Comment by jardakotesovec** [comment]
I did some testing with latest 4 and 5 version (using that new cool client option) and it seems working equally well to me. If you would have a chance to review it sometime soon that would be awesome, so it could be used also with version 5.
Thanks!

File: index.js [hunk 1]
```diff
@@ -40,34 +40,35 @@ exports.register = (server, options) => {
 
   // catch request errors/warnings/etc (default: only errors) and capture them with sentry
   server.events.on({ name: 'request', channels: opts.channels }, async (request, event) => {
-    // format a sentry event from the request and triggered event
-    const sentryEvent = await Sentry.Parsers.parseError(event.error);
-    Sentry.Handlers.parseRequest(sentryEvent, request.raw.req);
+    Sentry.withScope(scope => { // thus use a temp scope and re-assign it
+      scope.addEventProcessor(_sentryEvent => {
```
Comment (fiws) [inline index.js]: Why did you rename this to _sentryEvent?

btw: looks like an awesome PR, thanks!

Comment (guischdi) [reply index.js]: I think @jardakotesovec used `_sentryEvent` as a temporary handle for parsing it two lines below.
The parsed `sentryEvent` variable is used as before (manipulate `sentryEvent.request.url`, `sentryEvent.level`, etc).

File: index.js [hunk 2]
```diff
@@ -40,34 +40,35 @@ exports.register = (server, options) => {
...
+        // format a sentry event from the request and triggered event
+        const sentryEvent = Sentry.Handlers.parseRequest(_sentryEvent, request.raw.req);
```
Comment (fiws) [inline index.js]: This method modifies the passed event. At least it did so in sentry@4.
So could we remove the assignment, just call the method and rename `_sentryEvent` to `sentryEvent` ?

Comment (jardakotesovec) [reply index.js]: You are right, it modifies the original, so that assigning is not necessary. I still think that it makes sense to use return value as they do in their example and not rely on that its being just modified. That might change in future.  Reason for having _sentryEvent is just habit to avoid reassigning function arguments, which can have [performance impact](https://github.com/airbnb/javascript/issues/641#issuecomment-167827978), not sure if thats still the thing on latest node though.

**Comment by guischdi** [comment]
Thanks a lot @jardakotesovec 
I tested your changes and resolved a minor linting issue. The refactored feature seems to be working perfectly with @sentry/node@4 and @sentry/node@5!
